src/main/java/com/datatorrent/netlet/OptimizedEventLoop.java

```diff
@@ -75,9 +75,9 @@ public boolean add(SelectionKey key)
       }
       keys[pos++] = key;
       if (pos == keys.length) {
-        SelectionKey[] keys = new SelectionKey[this.keys.length << 1];
-        System.arraycopy(this.keys, 0, keys, 0, pos);
-        this.keys = keys;
+        SelectionKey[] lKeys = new SelectionKey[this.keys.length << 1];
```

Comment:
yes. I think that's better plan. How did you figure out if there are variables which are shadowing other variables?